mesonbuild/compilers/c.py

```diff
@@ -839,6 +843,20 @@ def get_library_naming(self, env, libtype, strict=False):
             raise AssertionError('BUG: unknown libtype {!r}'.format(libtype))
         return prefixes, suffixes
 
+    def verify_libtype(self, filename: Path, libtype: str):
+        # FIXME: shared-static and static-shared return the same for now
+        if libtype in ('shared-static', 'static-shared'):
+            return True
+        proc = subprocess.Popen(['lib', '/list', filename], stdout=subprocess.PIPE)
+        stdout, _ = proc.communicate()
+        stdout = stdout.decode('utf-8')
+        shlibname = filename.with_suffix('.dll').name
+        if libtype == 'static':
+            return not shlibname in stdout
```

Comment:
__[Flake8]__

[E713] test for membership should be 'not in'


<!-- Comment by [Sider](https://sider.review) -->